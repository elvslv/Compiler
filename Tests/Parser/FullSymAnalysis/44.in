var i, j, k, f: integer;
type tt = real;
procedure proc(m, n: integer);
type tt = integer;
var i, k: tt;
const m = 12;
begin
	if m then
		n := 7;
	for i := 1 to m do
		k := k + 1;
	f := k;
end;
begin
	proc(i, j);
	proc(i + 7, k);
end.