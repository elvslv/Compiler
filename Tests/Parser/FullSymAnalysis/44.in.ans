var I : INTEGER;
var J : INTEGER;
var K : INTEGER;
var F : INTEGER;
type TT = REAL;

procedure PROC(var M : INTEGER;var N : INTEGER;);
type TT = INTEGER;
var I : TT;
var K : TT;
const M = 12;

begin                
|                    
|___if               
|   |                
|   |___M            
|   |                
|   then             
|   |                
|   |___begin        
|       |            
|       |___:=       
|       |   |        
|       |   |___N    
|       |   |        
|       |   |___7    
|       |            
|       end          
|                    
|___for              
|   |                
|   |___:=           
|   |   |            
|   |   |___I        
|   |   |            
|   |   |___1        
|   |                
|   to               
|   |                
|   |___M            
|   |                
|   do               
|   |                
|   |___begin        
|       |            
|       |___:=       
|       |   |        
|       |   |___K    
|       |   |        
|       |   |___+    
|       |       |    
|       |       |___K
|       |       |    
|       |       |___1
|       |            
|       end          
|                    
|___:=               
|   |                
|   |___F            
|   |                
|   |___K            
|                    
end                  


begin                
|                    
|___()               
|   |                
|   |___PROC         
|   |                
|   |___I            
|   |                
|   |___J            
|                    
|___()               
|   |                
|   |___PROC         
|   |                
|   |___+            
|   |   |            
|   |   |___I        
|   |   |            
|   |   |___7        
|   |                
|   |___K            
|                    
end                  
