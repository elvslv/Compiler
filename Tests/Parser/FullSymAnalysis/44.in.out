var F : INTEGER;
var I : INTEGER;
var J : INTEGER;
var K : INTEGER;
procedure PROC(var M : INTEGER;var N : INTEGER;);
begin                
|                    
|___if               
|   |                
|   |___M            
|   |                
|   then             
|   |                
|   |___begin        
|       |            
|       |___:=       
|       |   |        
|       |   |___N    
|       |   |        
|       |   |___7    
|       |            
|       end          
|                    
|___for              
|   |                
|   |___:=           
|   |   |            
|   |   |___I        
|   |   |            
|   |   |___1        
|   |                
|   to               
|   |                
|   |___M            
|   |                
|   do               
|   |                
|   |___begin        
|       |            
|       |___:=       
|       |   |        
|       |   |___K    
|       |   |        
|       |   |___+    
|       |       |    
|       |       |___K
|       |       |    
|       |       |___1
|       |            
|       end          
|                    
|___:=               
|   |                
|   |___F            
|   |                
|   |___K            
|                    
end                  

type TT = REAL

begin                
|                    
|___()               
|   |                
|   |___PROC         
|   |                
|   |___I            
|   |                
|   |___J            
|                    
|___()               
|   |                
|   |___PROC         
|   |                
|   |___+            
|   |   |            
|   |   |___I        
|   |   |            
|   |   |___7        
|   |                
|   |___K            
|                    
end                  
